src/uct/sm/knem/knem_md.c

```diff
@@ -8,17 +8,41 @@
 #include "knem_md.h"
 #include "knem_io.h"
 
-ucs_status_t uct_knem_md_query(uct_md_h md, uct_md_attr_t *md_attr)
+#include <ucs/arch/cpu.h>
+
+static ucs_config_field_t uct_knem_md_config_table[] = {
+  {"", "", NULL,
+   ucs_offsetof(uct_knem_md_config_t, super), UCS_CONFIG_TYPE_TABLE(uct_md_config_table)},
+
+  {"RCACHE", "try", "Enable using memory registration cache",
+   ucs_offsetof(uct_knem_md_config_t, rcache.enable), UCS_CONFIG_TYPE_TERNARY},
+
+  {"RCACHE_ADDR_ALIGN", UCS_PP_MAKE_STRING(UCS_SYS_CACHE_LINE_SIZE),
```

Comment:
i would suggest to export common config struct to UCS to avoid code duplication for RCACHE parameters between IB and KNEM

Reply:
there is only one parent - uct_md_t... add rcache there?

Reply:
config infra supports "multiple inheritance" :) see example in https://github.com/openucx/ucx/blob/master/src/uct/ib/ud/base/ud_iface.c#L578

Reply:
done